plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// GRPCClient implements shared.StoragePlugin and reads/writes spans and dependencies
+type GRPCClient struct {
+	readerClient     storage_v1.SpanReaderPluginClient
+	writerClient     storage_v1.SpanWriterPluginClient
+	depsReaderClient storage_v1.DependenciesReaderPluginClient
+}
+
+// DependencyReader implements shared.StoragePlugin.
```

Comment:
If GRPCClient doesn't implement StoragePlugin then go-plugin fails to type assert the plugin on creation.